db/database.go

```diff
@@ -943,11 +943,11 @@ func (db *Database) UpdateAllDocChannels(doCurrentDocs bool, doImportDocs bool)
 		}
 		var err error
 		if db.UseXattrs() {
-			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte) (raw []byte, rawXattr []byte, err error) {
+			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, err error) {
```

Comment:
This is hard to mentally parse and also getting pretty horizontally wide.  What about pulling out the callback function?

```
callback := func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, err error) {
...
}
err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, callback)

```